SolidStateGroup/bullet-train-api
# Limit invite users according with org max seats

Thanks for submitting a PR! Please check the boxes below:


## Changes

### Backend:
Validates the organisation seats used vs max seats on the `GET api/v1/organisations/{pk}/invite-links` and `POST api/v1/users/join/#` endpoints. Throws HTTP Error 403.

### Frontend
- Handles `organisation reached max seats` error on API `get-invite-links` endpoint.
- Show max seats reached message.
<img width="890" alt="Screenshot 2023-06-09 at 01 10 00" src="https://github.com/Flagsmith/flagsmith/assets/41410593/b9c91fff-325a-4810-b985-7f1a87a06283">

- On sign-up from invite link, catches API error when organisation reached max seats 
<img width="1344" alt="Screen Shot 2023-06-22 at 15 34 09" src="https://github.com/Flagsmith/flagsmith/assets/41410593/cdc673d0-4a19-4a4b-94ee-ddd49516ab8c">

## How did you test this code?

Manual Locally
- In the local database choose an organisation in `organisations_subscription` table, and set `max_seat` value to N, being N the max number of seats you want to test.
- Using the invite link create N users. Copy the invite link to re-use it.
- See that when the number of users is N the invite link is not displayed and a warning message is shown instead. 
- See that for the n + 1 attempt you are not able to sign-up using the existing invite link.

E2E tests should be considered for the scenarios covered by the new behavior implemented.
